pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ApexTreeBuilder.java

```diff
@@ -271,6 +279,71 @@ public ApexTreeBuilder(String sourceCode) {
         return node;
     }
 
+    private void buildFormalComment(AstNode node) {
+        if (parents.peek() == node) {
+            ApexNode<?> parent = (ApexNode<?>) nodes.peek();
+            TokenLocation tokenLocation = getApexDocTokenLocation(getApexDocIndex(parent));
+            if (tokenLocation != null) {
+                ASTFormalComment comment = new ASTFormalComment(tokenLocation.token);
+                comment.calculateLineNumbers(sourceCodePositioner, tokenLocation.index,
+                        tokenLocation.index + tokenLocation.token.length());
+                parent.jjtAddChild(comment, 0);
+                comment.jjtSetParent(parent);
+            }
+        }
+    }
+
+    private int getApexDocIndex(ApexNode<?> node) {
+        ASTAnnotation annotation = node.getFirstDescendantOfType(ASTAnnotation.class);
+        ApexNode<?> firstNode = annotation == null ? node : annotation;
+        int index = firstNode.getNode().getLoc().getStartIndex();
+        return sourceCode.lastIndexOf('\n', index);
+    }
+
+    private TokenLocation getApexDocTokenLocation(int index) {
+        TokenLocation last = null;
+        for (TokenLocation location : tokenLocations) {
+            if (location.index >= index) {
```

Comment:
the `tokenLocations` list is sorted, we can probably be smarter about this search